apps/prairielearn/src/sync/fromDisk/courseInstances.ts

```diff
@@ -38,6 +41,41 @@ export async function sync(
   courseId: string,
   courseData: CourseData,
 ): Promise<Record<string, string>> {
+  if (config.checkInstitutionsOnSync) {
```

Comment:
Should we do the thing we do elsewhere where we only add warnings for course instances which will be accessible in the future?

Reply:
I think so? That will certainly cut down on unnecessary errors for people.

Reply:
I'd suggest applying it directly to the access rule where this is validated. This makes the test simpler (endDate null or in future), and the alternative of checking all access rules does not provide meaningful gains.

Reply:
@jonatanschroeder I'm not sure what you mean by "I'd suggest applying it directly to the access rule where this is validated", can you clarify?

Reply:
I added limiting by end date in 6833160. I applied the end date checks to each individual access rule rather than the course instance as a whole, which I think is what Jonatan was suggesting above.

Reply:
> @jonatanschroeder I'm not sure what you mean by "I'd suggest applying it directly to the access rule where this is validated", can you clarify?

I mean that, for example, if a course instance has two access rules, one with an institution and a past date, and one without an institution and a future date, this should not trigger an error. The institution error happens in an access rule that is no longer active. And this matches what you have done in the recent commit.